docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md

```diff
@@ -0,0 +1,179 @@
+                "content-type": "application/json"
+            },
+            "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/invoke",
+            "request_body": "{ \"texts\": ${parameters.texts}, \"truncate\": \"${parameters.truncate}\", \"input_type\": \"${parameters.input_type}\" }",
+            "pre_process_function": "connector.pre_process.cohere.embedding",
+            "post_process_function": "connector.post_process.cohere.embedding"
+        }
+    ]
+}
+```
+
+If using the AWS Opensearch Service, you can provide an IAM role arn that allows access to the bedrock service.
+Refer to this [AWS doc](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ml-amazon-connector.html)
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "Amazon Bedrock",
+    "description": "Test connector for Amazon Bedrock",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "credential": {
+        "roleArn": "<PLEASE ADD YOUR AWS ROLE ARN HERE>"
+    },
+    "parameters": {
+      "region": "<PLEASE ADD YOUR AWS REGION HERE>",
+      "service_name": "bedrock",
+      "truncate": "<NONE|START|END>",
+      "input_type": "<search_document|search_query|classification|clustering>",
+      "model": "cohere.embed-english-v3"
+    },
+    "actions": [
+        {
+          "action_type": "predict",
+          "method": "POST",
+          "headers": {
+            "x-amz-content-sha256": "required",
+            "content-type": "application/json"
+          },
+          "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/invoke",
+          "request_body": "{ \"texts\": ${parameters.texts}, \"truncate\": \"${parameters.truncate}\", \"input_type\": \"${parameters.input_type}\" }",
+          "pre_process_function": "connector.pre_process.cohere.embedding",
+          "post_process_function": "connector.post_process.cohere.embedding"
+        }
+    ]
+}
+```
+
+Sample response:
+```json
+{
+  "connector_id": "hijwwZABNrAVdFa9prf7"
+}
+```
+
+## 3. Create model group:
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "remote_model_group_claude3",
+    "description": "This is an example description"
+}
+```
+
+Sample response:
+```json
+{
+  "model_group_id": "IMobmY8B8aiZvtEZeO_i",
+  "status": "CREATED"
+}
+```
+
+## 4. Register model to model group & deploy model:
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "cohere.embed-english-v3",
+    "function_name": "remote",
+    "model_group_id": "IMobmY8B8aiZvtEZeO_i",
+    "description": "cohere embed-english v3 model",
+    "connector_id": "hijwwZABNrAVdFa9prf7"
+}
+```
+
+Sample response:
+```json
+{
+  "task_id": "rMormY8B8aiZvtEZIO_j",
+  "status": "CREATED",
+  "model_id": "lyjxwZABNrAVdFa9zrcZ"
+}
+```
+
+## 5. Test model inference
+
+```json
+POST /_plugins/_ml/models/rcormY8B8aiZvtEZIe89/_predict
```

Comment:
Can you also try if this API can work ?  neural-search requires this API can work

```
POST /_plugins/_ml/_predict/text_embedding/<model_id>
{
  "text_docs":[ "today is sunny", "today is sunny"],
  "return_number": true,
  "target_response": ["sentence_embedding"]
}
```

Reply:
Yes this is also verified too. It invokes successfully.